es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java

```diff
@@ -748,6 +754,47 @@ private void addSortOptionToSearchRequest(SearchRequestBuilder searchRequestBuil
             .collect(Collectors.toCollection(LinkedList::new));
     }
 
+    @Override
+    public void groomWorkflowsAndTasks() {
+        // Groom oldest archived workflows by batch size
+        QueryBuilder wfQuery = QueryBuilders.existsQuery("archived");
+        long workflowsGroomed = groomDocs(indexName, wfQuery, WORKFLOW_DOC_TYPE, "endTime");
+
+        // Groom obsolete tasks that are staying for more than a day
+        int daysToKeepTasks = 1;
+        DateTime dateTime = new DateTime();
+        QueryBuilder taskQuery = QueryBuilders.rangeQuery("updateTime").lt(dateTime.minusDays(daysToKeepTasks));
+
+        long tasksGroomed = groomDocs(indexName, taskQuery, TASK_DOC_TYPE, "updateTime");
+
+        logger.info("Groomed {} workflows and {} tasks", workflowsGroomed, tasksGroomed);
+    }
+
+    public long groomDocs(String indexName, QueryBuilder q, String docType, String sortField) {
+        int batchSize = config.getElasticSearchGroomingBatchSize();
```

Comment:
If I read batchsize, my assumption is that if we fetch 1000 records and batchsize is 100 then we eventually delete all 1000 however in a batch of 100.
seems here this is not the case, we should just name the variable as `size` rather `batchsize`.
I think there should be separate batch size for Task and Workflow as Tasks are far more than Workflows.